src/kernel/include/thd.h

```diff
@@ -472,17 +498,30 @@ curr_invstk_top(struct cos_cpu_local_info *cos_info)
 	return cos_info->invstk_top;
 }
 
+/* 
+ * This function aims to output the "in_fault" flag and return comp_info with the bit in captbl reset.
+ * Refer to the comments of comp_invstk_info to see more about this design.
+ */
 static inline struct comp_info *
-thd_invstk_current(struct thread *curr_thd, unsigned long *ip, unsigned long *sp, struct cos_cpu_local_info *cos_info)
+thd_invstk_current_fault(struct thread *curr_thd, unsigned long *ip, unsigned long *sp, unsigned long *in_fault, struct cos_cpu_local_info *cos_info)
```

Comment:
Signature is so jumbled.. You'd want to order the "INPUT args" and "OUTPUT args" correctly..